Carthage support

rad3ks opened the pull request:
Carthage depends on dynamic frameworks and since SwiftyUserDefaults was meant to be used with CocoaPods you didn't need any `xcodproj`. To support Carthage those targets were needed so I made xcodproj that has two targets for each iOS and OSX. I hope this isn't a problem. CocoaPods support should remain unchanged.

I also moved tests from `main.swift` to test file and substituted `assert` with `XCTAssert`, so now tests can be run for each platform individually.

radex commented:
Thanks @rad3ks, but #14 already added Carthage support (it just took me a while to get it merged).

Could you elaborate about why `XCTAssert` is better than `assert`? I mean, I know the latter doesn't integrate with Xcode, but for the purposes of this project, it works fine. What do you mean about running for "each platform individually"?

rad3ks commented:
Oh thanks, didn't check if there is already a pending PR with Carthage support.

I used `XCTAssert` because I moved tests to `XCTestCase`, this is the only reason.

For each platform individually, because I added separate framework targets for iOS and OS X. This was just a quick PR so I can move on with my own job (need only iOS platform) and I couldn't figure out how to build all architectures from a single framework target (had some issues - don't remember specifics). 

Thanks for merging that other PR :+1:
